### ISSUE TITLE
Quantity Values on Variants don't properly inherit Data of parent when a Default Unit is set

### ISSUE THREAD

**Issue by fweisflug** [pimcore/pimcore#6754]
# Bug Report
Quantity Values on Variants don't properly inherit Data of parent when a Default Unit is set

### Expected behavior
If a Variant of an Object is created with Quantity Values that have a Default Unit set, that Quantity Value should Properly inherit the data of the parent.

### Actual behavior
The effected Quantity Values is overwritten and isn't inherited (Green Triangle top left and greyed out).

### Steps to reproduce
1. Add a Quantity Value Unit Definition
2. Create a Class (Inheritance and Variants turned on) with two Quantity Values with the same Unit, but set the Default Unit of one of them.
3. Create an Object of that Class and fill out both Quantity Values, Save and Publish
4. Right click on the newly created Object and click Add Variant
5. The quantity Value without a Default Unit set properly Inherits the Value, the other looks like it was already edited/overwritten in the variant (no green triangle, not greyed out)

Tested on https://demo.pimcore.fun and how it looks like
![Screenshot 2020-07-08 at 15 20 16](https://user-images.githubusercontent.com/49750472/86923642-98891b00-c12e-11ea-94d1-46623653223e.png)
![Screenshot 2020-07-08 at 15 20 22](https://user-images.githubusercontent.com/49750472/86923645-9921b180-c12e-11ea-8337-63cfc8c8a4c1.png)

### Example Export of Test Class
[class_BrickTest_export.txt](https://github.com/pimcore/pimcore/files/4890850/class_BrickTest_export.txt)

**Comment by weisswurstkanone** [comment]
This is wanted behavior. See
https://github.com/pimcore/pimcore/blob/672b674763688b1697ffced1ab5d8e1ae7670584/models/DataObject/Traits/DefaultValueTrait.php#L123

Having a default value or unit basically indicates that we want to work with default values.
However, if inheritance is enabled we take the value from the parent (if it is not empty) and use this one instead. So this scenario would break the inheritance.
If the parent doesn't have a value then the configured default is used.
If no default value is configured then inheritance works as usual (as you see in "testWorks")

**Comment by weisswurstkanone** [comment]
already discussed here: https://github.com/pimcore/pimcore/issues/6376

**Comment by fweisflug** [comment]
just to make sure. so the behaviour if data is persisted to the Variant and which one is persisted on the variant on creation depends on whether a Default Unit is set or not?

assuming a Class with one Quantity Value named "testValue" with a Default Unit set to "m". we create a new Object of that Class and set "testValue" to "4".

currently if we create a Variant of this Object, the "testValue" of "4" (and its Unit "m") is copied from the Parent to the Variant and persisted on the Variant. Changes on the Parent do not affect the values (shown) on the Variant. There is no real inheritance (no green triangle/not greyed out) besides the initial copy.

```
ParentObejct:
 | testValue: 4 m
 |
 |----> Variant1:
            testValue: 4 m 
```

what i would expect, is that no Data is copied from the Parent to the Variant and the values are inherited from the Parent, eg no data is persisted on the Variant but instead inherited from the Parent (green triangle/greyed out). Changes on the Parent do affect the values shown on the Variant.

```
ParentObejct:
 | testValue: 4 m
 |
 |----> Variant1:
            testValue: empty (inherited from Parent 4 m)
```

Question here is, why does the a set Default Unit change the behaviour of the inheritance, since without a Default Unit set my expected behaviour aligns with the current one?


> https://github.com/pimcore/pimcore/blob/735282d92b7097ca2687f8881ed06075fbe116a5/doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/README.md#L158


this readme seems to be from an old commit (735282d on 20 Apr). the same readme on master (https://github.com/pimcore/pimcore/blob/master/doc/Development_Documentation/05_Objects/01_Object_Classes/01_Data_Types/README.md) adds:

> 1. **No inheritance:** default value is persisted to store/query table on create
> 2. **With inheritance and NO parent value:** default value is persisted to store/query table on create
> 3. **With inheritance and existing parent value:** no value is persisted to store table, inherited value is persisted to query table, inheritance is active
> 

doesn't point 3 imply that the actual behaviour should be like the wanted behaviour i described and not how it currently behaves?

**Comment by fweisflug** [comment]
i just tried out the master branch with those commits included from #6400. with that everything works like i would expect it to.
